Sources/FluentKit/Concurrency/AsyncModelMiddleware.swift

```diff
@@ -14,24 +14,32 @@ public protocol AsyncModelMiddleware: AnyModelMiddleware {
 
 @available(macOS 12, iOS 15, watchOS 8, tvOS 15, *)
 extension AsyncModelMiddleware {
-    public func handle(_ event: ModelEvent, _ model: AnyModel, on db: Database, chainingTo next: AnyAsyncModelResponder) async throws {
-        guard let modelType = model as? Model else {
-            try await next.handle(event, model, on: db)
-            return
-        }
-        
-        switch event {
-        case .create:
-            try await self.create(model: modelType, on: db, next: next)
-        case .update:
-            try await self.update(model: modelType, on: db, next: next)
-        case .delete(let force):
-            try await self.delete(model: modelType, force: force, on: db, next: next)
-        case .softDelete:
-            try await self.softDelete(model: modelType, on: db, next: next)
-        case .restore:
-            try await self.restore(model: modelType, on: db, next: next)
+    public func handle(_ event: ModelEvent, _ model: AnyModel, on db: Database, chainingTo next: AnyModelResponder) -> EventLoopFuture<Void> {
+        let promise = db.eventLoop.makePromise(of: Void.self)
```

Comment:
Slightly shorter version:

```suggestion
        return db.eventLoop.performWithTask { ...
```

Reply:
This would require bumping the base required version of `async-kit` to ensure the utility is available, which increases the semver-ness of the change.

Reply:
FluentKit doesn't export AsyncKit (🎉🎉🎉 there's at least one!) so we could bump it in the manifest in a patch version. But I'm going to keep it as is as it's one less place to add a dependency and it keeps it in line with the rest of our concurrency stuff